bindings_generator/src/special_methods.rs

```diff
@@ -135,243 +150,245 @@ pub fn generate_free_impl(
     api: &Api,
     class: &GodotClass,
 ) -> GeneratorResult {
-    if class.instanciable && !class.is_pointer_safe() {
-        writeln!(
-            output,
-            r#"
-impl Free for {name} {{
-    #[inline]
-    unsafe fn godot_free(self) {{ self.free() }}
-}}"#,
-            name = class.name,
-        )?;
-    }
+    let class_name = format_ident!("{}", class.name);
+    let free_output = if class.instantiable && !class.is_pointer_safe() {
+        quote! {
+            impl Free for #class_name {
+                #[inline]
+                unsafe fn godot_free(self) { self.free() }
+            }
+        }
+    } else {
+        Default::default()
+    };
 
-    if class.name == "Node" || api.class_inherits(&class, "Node") {
-        writeln!(
-            output,
-            r#"
-impl QueueFree for {name} {{
-    #[inline]
-    unsafe fn godot_queue_free(&mut self) {{ self.queue_free() }}
-}}"#,
-            name = class.name,
-        )?;
-    }
+    let queue_free_output = if class.name == "Node" || api.class_inherits(&class, "Node") {
+        quote! {
+            impl QueueFree for #class_name {
+                #[inline]
+                unsafe fn godot_queue_free(&mut self) { self.queue_free() }
+            }
+        }
+    } else {
+        Default::default()
+    };
+
+    let code = quote! {
+        #free_output
+        #queue_free_output
+    };
+
+    generated_at!(output);
+    write!(output, "{}", code)?;
 
     Ok(())
 }
 
-pub fn generate_singleton_getter(output: &mut impl Write, class: &GodotClass) -> GeneratorResult {
+pub fn generate_singleton_getter(class: &GodotClass) -> TokenStream {
+    if !class.singleton {
+        return Default::default();
+    }
+
```

Comment:
This should just be an assert. It's better to add a branch where this is called.